tests/routes/cbauthentication_tests/cb_login.test.js

```diff
@@ -5,44 +5,51 @@ const { getConfig } = require('../../../config');
 
 const config = getConfig(process.env.NODE_ENV);
 
-test('POST /api/cb/login | authentication successful', (t) => {
-  const successPayload = { formEmail: 'findmyfroggy@frogfinders.com', formPswd: 'Funnyfingers11!' };
+test('POST /api/cb/login', (tape) => {
   const app = createApp(config);
+  const dbConnection = app.get('client:psql');
 
-  request(app)
-    .post('/api/cb/login')
-    .send(successPayload)
-    .expect(200)
-    .expect('Content-Type', /json/)
-    .end((err, res) => {
-      t.ok(res.body.success, 'cb successfully logged in with correct password');
-      t.end();
-    });
-});
+  tape.test('POST /api/cb/login | authentication successful', (t) => {
+    const successPayload = { formEmail: 'findmyfroggy@frogfinders.com', formPswd: 'Funnyfingers11!' };
 
-test('POST /api/cb/login | authentication unsuccessful', (t) => {
-  t.plan(4);
-  const failPayload = { formEmail: 'findmyfroggy@frogfinders.com', formPswd: 'password123' };
-  const emptyPayload = { formEmail: '', formPswd: '' };
-  const app = createApp(config);
+    request(app)
+      .post('/api/cb/login')
+      .send(successPayload)
+      .expect(200)
+      .expect('Content-Type', /json/)
+      .end((err, res) => {
+        t.ok(res.body.success, 'cb successfully logged in with correct password');
+        t.end();
+      });
+  });
+
+  tape.test('POST /api/cb/login | authentication unsuccessful', (t) => {
+    t.plan(4);
+
+    const failPayload = { formEmail: 'findmyfroggy@frogfinders.com', formPswd: 'password123' };
+    const emptyPayload = { formEmail: '', formPswd: '' };
 
-  request(app)
-    .post('/api/cb/login')
-    .send(failPayload)
-    .expect(401)
-    .expect('Content-Type', /json/)
-    .end((err, res) => {
-      t.notOk(err, err || 'Passes supertest expect criteria');
-      t.notOk(res.body.success, 'cb failed to log in with incorrect password');
-    });
-
-  request(app)
-    .post('/api/cb/login')
-    .send(emptyPayload)
-    .expect(400)
-    .expect('Content-Type', /json/)
-    .end((err, res) => {
-      t.notOk(err, err || 'Passes supertest expect criteria');
-      t.notOk(res.body.success, 'cb failed to log in with no submitted data');
-    });
+    request(app)
+      .post('/api/cb/login')
+      .send(failPayload)
+      .expect(401)
+      .expect('Content-Type', /json/)
+      .end((err, res) => {
+        t.notOk(err, err || 'Passes supertest expect criteria');
+        t.notOk(res.body.success, 'cb failed to log in with incorrect password');
+      });
+
+    request(app)
+      .post('/api/cb/login')
+      .send(emptyPayload)
+      .expect(400)
+      .expect('Content-Type', /json/)
+      .end((err, res) => {
+        t.notOk(err, err || 'Passes supertest expect criteria');
+        t.notOk(res.body.success, 'cb failed to log in with no submitted data');
+      });
+  });
+
+  tape.test('POST /api/cb/login | teardown', (t) => dbConnection.end(t.end));
```

Comment:
Ooo what does this do?

Reply:
Why are we using the `Pool` here rather than `Client`?

Reply:
In this test, we're creating a new instance of the application, which creates an instance of `pg.Pool`. To properly teardown the test, we need to close the connections created by `pg.Pool`.

If we were testing the query functions directly, or anything that accepted a `pg.Pool` instance as an argument, we would probably use `pg.Client` instead, unless the code used some part of the API specific to `pg.Pool`.